PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt

```diff
@@ -279,8 +294,26 @@ open class SparkSubmissionContentPanel(private val myProject: Project, val type:
         toolTipText = "Files to be placed on the java classpath; The path needs to be an Azure Blob Storage Path (path started with wasb://); Multiple paths should be split by semicolon (;)"
     }
 
-    private val referencedJarsTextField: ExpandableTextField = ExpandableTextField(ParametersListUtil.COLON_LINE_PARSER, ParametersListUtil.COLON_LINE_JOINER).apply {
-        toolTipText = refJarsPrompt.toolTipText
+    private val referencedJarsTextField: TextFieldWithBrowseButton = TextFieldWithBrowseButton(ExpandableTextField(ParametersListUtil.COLON_LINE_PARSER, ParametersListUtil.COLON_LINE_JOINER).apply {
+        toolTipText = "Artifact from remote storage account."
+    }).apply {
+        textField.document.addDocumentListener(documentValidationListener)
+        button.addActionListener {
+            val uploadRootPath = storageWithUploadPathPanel.uploadPathField.text.trim()
+                    .replace("/${Constants.submissionFolder}/", "")
```

Comment:
We may have to limit the matching to be end of string and with/without `/`

```suggestion
                    .replace("/${Constants.submissionFolder}/?$", "")
```